enhancements/insights/config-api.md

```diff
@@ -0,0 +1,286 @@
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=false
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling some gatherers can look like `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatheringTime is the last time when Insights gathering finished.
+	LastGatheringTime metav1.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status GatherStatus `json:"status,omitempty"`
+	// List of active gatherers (and their statuses) in the last gathering.
+	GathererStatuses []GathererStatus `json:"gathererStatuses,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate,omitempty"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type GathererStatus struct {
+	// Name is the name of the gatherer.
+	Name string `json:"name"`
```

Comment:
It's about the gatherers, not the gathering functions, right?

Reply:
I think it's more about gathering functions here. Do you think we should choose a better name/naming?

Reply:
Yes, I'd make it complaint with terminology in IO

Reply:
We agreed with Serhii that the naming is OK.